megamek/src/megamek/common/alphaStrike/conversion/ASConvInfantryDamageConverter.java

```diff
@@ -42,27 +42,34 @@ protected ASConvInfantryDamageConverter(Entity entity, AlphaStrikeElement elemen
 
     @Override
     protected void processDamage() {
-        report.addEmptyLine();
-        report.addLine("--- Damage:", "");
-        int baseRange = 0;
-        if ((infantry.getSecondaryWeapon() != null) && (infantry.getSecondaryWeaponsPerSquad() >= 2)) {
-            baseRange = infantry.getSecondaryWeapon().getInfantryRange();
-        } else if (infantry.getPrimaryWeapon() != null) {
-            baseRange = infantry.getPrimaryWeapon().getInfantryRange();
-        }
-        int range = baseRange * 3;
-        finalSDamage = ASDamage.createDualRoundedUp(getConvInfantryStandardDamage());
-        String maxRangeText = "Range: S";
-        if (range > 3) {
-            finalMDamage = finalSDamage;
-            maxRangeText = "Ranges: S, M";
-        }
-        if (range > 15) {
-            finalLDamage = finalSDamage;
-            maxRangeText = "Ranges: S, M, L";
+
+        if (infantry.hasFieldWeapon()) {
+            processSDamage();
+            processMDamage();
+            processLDamage();
+            processFrontSpecialDamage(AC);
+            processFrontSpecialDamage(FLK);
+        } else {
+            int baseRange = 0;
+            if ((infantry.getSecondaryWeapon() != null) && (infantry.getSecondaryWeaponsPerSquad() >= 2)) {
+                baseRange = infantry.getSecondaryWeapon().getInfantryRange();
+            } else if (infantry.getPrimaryWeapon() != null) {
+                baseRange = infantry.getPrimaryWeapon().getInfantryRange();
+            }
+            int range = baseRange * 3;
+            finalSDamage = ASDamage.createDualRoundedUp(getConvInfantryStandardDamage());
+            String maxRangeText = "Range: S";
+            if (range > 3) {
+                finalMDamage = finalSDamage;
+                maxRangeText = "Ranges: S, M";
+            }
+            if (range > 15) {
+                finalLDamage = finalSDamage;
+                maxRangeText = "Ranges: S, M, L";
+            }
```

Comment:
```suggestion
            final String maxRangeText;
            if (range > 15)
                finalLDamage = finalSDamage;
                finalMDamage = finalSDamage;
                maxRangeText = "Ranges: S, M, L";
            } else if (range > 3) {
                finalMDamage = finalSDamage;
                maxRangeText = "Ranges: S, M";
            } else {
                maxRangeText  = "Range: S";
            }
```

Reply:
This version (the first two parts specifically) prevents misreads, and should range > 15 be assigning the finalMDamage?

Reply:
Yes I'll do that. Infantry damage (disregarding field guns) is the same for S/M/L but extends to either S only, S/M or S/M/L based on the range of the employed infantry weapon. So when the weapon has 20 range, the S dmg value is used for M and L too.